Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift

```diff
@@ -0,0 +1,136 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+/// A control that displays an editable text interface for search inputs.
+///
+/// Use `inputState(_ state: State)` to change the state of the text field.
+public struct BPKSearchInputSummary: View {
+    public enum InputPrefix {
+        case text(String)
+        case icon
+    }
+    
+    struct Icon {
+        let icon: BPKIcon
+        let color: BPKColor
+    }
+    
+    @Binding private var text: String
+    private let placeholder: String
+    private let inputPrefix: InputPrefix
+    private var state: State = .default
+    
+    @FocusState private var focused: Bool
+
+    /// Creates a `BPKSearchInputSummary`.
+    ///
+    /// - Parameters:
+    ///   - placeholder: The placeholder text to display when the text field is empty.
+    ///   - inputPrefix: The prefix which would be displayed on the left of text input
+    ///   - text: The text to display in the text field.
+    public init(
+        placeholder: String = "",
+        inputPrefix: InputPrefix = .icon,
+        _ text: Binding<String>
+    ) {
+        self.placeholder = placeholder
+        self.inputPrefix = inputPrefix
+        self._text = text
+    }
+    
+    public var body: some View {
+        HStack {
+            prefixView
+                .accessibilityHidden(true)
+            TextField(placeholder, text: $text)
+                .font(style: .bodyDefault)
+                .foregroundColor(state.textColor)
+                .disabled(state.isDisabled)
+                .focused($focused)
+            accessory
+        }
+        .padding(.md)
+        .background(.surfaceDefaultColor)
+        .clipShape(RoundedRectangle(cornerRadius: .sm))
+        .outline(focused ? .textLinkColor : state.borderColor, cornerRadius: .sm, lineWidth: focused ? 2.0 : 1.0)
+    }
+    
+    @ViewBuilder
+    private var accessory: some View {
+        if let icon = state.icon {
+            if case let .clear(accessibilityLabel, action) = state {
+                Button(action: action) {
+                    BPKIconView(icon.icon)
+                        .foregroundColor(icon.color)
+                }
+                    .accessibilityLabel(accessibilityLabel)
+            } else {
+                BPKIconView(icon.icon)
+                    .foregroundColor(icon.color)
+                    .accessibilityHidden(true)
+            }
+        }
+    }
+    
+    @ViewBuilder
+    private var prefixView: some View {
+        switch inputPrefix {
+        case .text(let prefixText):
+            BPKText(prefixText, style: .bodyDefault)
+                .foregroundColor(.textSecondaryColor)
+        case .icon:
+            BPKIconView(BPKIcon(name: "search"))
```

Comment:
the icon should be passed by the ones using this component, shouldnt be hardcoded within. also, you should use the constant for the icon rather than a string for it